x/gamm/types/pool.go

```diff
@@ -40,30 +42,28 @@ type PoolAccountI interface {
 }
 
 var (
-	// TODO: Add `GenesisAccount` type
-	_                         PoolAccountI = (*PoolAccount)(nil)
-	MaxUserSpecifiedWeight    sdk.Int      = sdk.NewIntFromUint64(1 << 20)
-	GuaranteedWeightPrecision int64        = 1 << 30
+	_                         PoolI   = (*Pool)(nil)
+	MaxUserSpecifiedWeight    sdk.Int = sdk.NewIntFromUint64(1 << 20)
+	GuaranteedWeightPrecision int64   = 1 << 30
 )
 
 func NewPoolAddress(poolId uint64) sdk.AccAddress {
-	return sdk.AccAddress(crypto.AddressHash(append(PoolAddressPrefix, sdk.Uint64ToBigEndian(poolId)...)))
+	return address.Module(ModuleName, sdk.Uint64ToBigEndian(poolId))
```

Comment:
Lets add more domain separation here. I suggest making the RHS `"Pool" + sdk.Uint64ToBigEndian(poolId)`

Reply:
(The concern is that there could one day be other addresses in this module, and then we have a headache for ensuring that they can never conflict)